include/mlir/ExecutionEngine/ExecutionEngine.h

```diff
@@ -42,6 +45,28 @@ namespace impl {
 class OrcJIT;
 } // end namespace impl
 
+/// Object cache for MLIR JIT-compiled objects. We pass an instance of this
+/// class to OrcJit to retrieve the object generated for a single MLIR module.
+/// We only use it to dump JIT-compiled code to object files (see
+/// `dumpObjectFile`).
+class MLIRObjectCache : public llvm::ObjectCache {
+public:
+  void notifyObjectCompiled(const llvm::Module *M,
+                            llvm::MemoryBufferRef ObjBuffer) override;
+
+  std::unique_ptr<llvm::MemoryBuffer>
+  getObject(const llvm::Module *M) override {
+    // Not implemented.
```

Comment:
Is this function ever called in normal operation?  If not, I'd rather put llvm_unreachable("unimplemented");